plugins/importtext_procedure.cc

```diff
@@ -1119,10 +1124,16 @@ struct ImportTextProcedureWorkInstance
                                          0 /* todo: chunk ofs */);
 
             ExpressionValue nameStorage;
-            RowPath rowName(namedBound(row, nameStorage, GET_ALL)
-                                .toUtf8String());
-            row.rowName = &rowName;
+            RowPath rowName;
 
+            if (isNamedLineNumber) {
+                rowName = Path(actualLineNum);
+            }
+            else {
+                rowName = namedBound(row, nameStorage, GET_ALL).coerceToPath();
```

Comment:
And note, toUtf8String() is completely wrong to use there, as it doesn't allow for any structured row names.  In other words, this is the only correct way to do this.